lib/internal/modules/esm/translators.js

```diff
@@ -125,23 +137,106 @@ translators.set('module', async function moduleStrategy(url) {
 // Strategy for loading a node-style CommonJS module
 const isWindows = process.platform === 'win32';
 const winSepRegEx = /\//g;
-translators.set('commonjs', function commonjsStrategy(url, isMain) {
+translators.set('commonjs', async function commonjsStrategy(url, isMain) {
   debug(`Translating CJSModule ${url}`);
-  return new ModuleWrap(url, undefined, ['default'], function() {
+
+  let filename = internalURLModule.fileURLToPath(new URL(url));
+  if (isWindows)
+    filename = StringPrototypeReplace(filename, winSepRegEx, '\\');
+
+  // Apply CJS exports parser.
+  if (!cjsParserInitialized) {
+    await cjsInit();
+    cjsParserInitialized = true;
+  }
+  const { module, exportNames } = cjsPreparseModuleExports(filename);
+  const namesWithDefault = exportNames.has('default') ?
+    [...exportNames] : ['default', ...exportNames];
```

Comment:
```suggestion
    ArrayFrom(exportNames) : ArrayPrototypeConcat('default', exportNames);
```
not strictly necessary, but much more robust